### PR TITLE
Convert node statues into standalone objects

### PR DESCRIPTION
**Pull request by mrogers950**
This moves the NodeStatus fields into proper FileIntegrityNodeStatus objects in
order to better scale with lots of nodes.

https://issues.redhat.com/browse/CMP-616

### CODE REVIEW

File: pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go [hunk 1]
```diff
@@ -41,13 +41,16 @@ type FileIntegrityConfig struct {
...
+	Phase FileIntegrityStatusPhase `json:"phase,omitempty"`
 }
 
-// NodeStatus defines the status of a specific node
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// FileIntegrityNodeStatus defines the status of a specific node
 // +k8s:openapi-gen=true
-type NodeStatus struct {
+type FileIntegrityNodeStatus struct {
```
Comment (JAORMX) [inline pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go:51]: I think you forgot to add the new CRD to the commit.

File: pkg/controller/configmap/configmap_controller.go [hunk 2]
```diff
@@ -229,16 +230,16 @@ func (r *ReconcileConfigMap) handleIntegrityLog(cm *corev1.ConfigMap, logger log
...
 		}
 	} else {
-		status := fileintegrityv1alpha1.NodeStatus{
+		status := &fileintegrityv1alpha1.FileIntegrityNodeStatus{
 			Condition:     fileintegrityv1alpha1.NodeConditionSucceeded,
 			NodeName:      node,
 			LastProbeTime: cm.GetCreationTimestamp(),
 		}
-		if err := r.updateStatus(fi, status); err != nil {
+		if err := r.createOrUpdateNodeStatus(node, fi, status); err != nil {
```
Comment (JAORMX) [inline pkg/controller/configmap/configmap_controller.go:242]: To update conditions, we could start taking this into use https://github.com/operator-framework/operator-sdk/blob/master/pkg/status/conditions.go

File: pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go [hunk 3]
```diff
@@ -41,13 +41,16 @@ type FileIntegrityConfig struct {
...
+	metav1.TypeMeta          `json:",inline"`
+	metav1.ListMeta          `json:"metadata,omitempty"`
```
Comment (JAORMX) [inline pkg/apis/fileintegrity/v1alpha1/fileintegrity_types.go:53]: Also... should we take this condition object into use to extend this? https://github.com/operator-framework/operator-sdk/blob/master/pkg/status/conditions.go

File: pkg/common/constants.go [hunk 4]
```diff
@@ -19,6 +19,8 @@ const (
 	IntegrityConfigMapOwnerLabelKey = "file-integrity.openshift.io/owner"
 	// IntegrityConfigMapNodeLabelKey tells us from which node did the configmap come from
 	IntegrityConfigMapNodeLabelKey = "file-integrity.openshift.io/node"
+	// IntegrityConfigMapNodeLabelKey tells us from which node did the configmap come from
```
Comment (JAORMX) [inline pkg/common/constants.go:22]: s/configmap/status/

File: pkg/controller/configmap/configmap_controller.go [hunk 5]
```diff
@@ -195,13 +197,18 @@ func (r *ReconcileConfigMap) handleIntegrityLog(cm *corev1.ConfigMap, logger log
 			}
 		}
 
-		status := fileintegrityv1alpha1.NodeStatus{
+		status := &fileintegrityv1alpha1.FileIntegrityNodeStatus{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fi.Name + "-" + node + "-err",
```
Comment (JAORMX) [inline pkg/controller/configmap/configmap_controller.go:202]: Why add the result suffix (`-err`). Wouldn't it be better if we just use `<file integrity name> + "-" + <node name>` to identify the status, and just video what the latest status is?

**Comment by jhrozek** [comment]
@mrogers950 can you rebase atop master? I think the recently pushed PR to upgrade the go builders to 1.13 should clear the CI issues..

**Comment by mrogers950** [comment]
@jhrozek rebased

**Pull request merged**